client/src/components/DateTime/DateRange.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+
+import { View } from 'react-native';
+import { DateSelect } from './';
+import { NOT_SPECIFIED, AVAILABLE, OUT_OF_RANGE } from '../../constants';
+
+import styles from './styles';
+
+const DateRange = ({ timeSegments, startTime, endTime, selectedDays, onSelect }) => {
+  const startDay = moment.unix(startTime).startOf('day');
+  const endDay = moment.unix(endTime).endOf('day');
+
+  const startOfWeek = startDay.startOf('isoWeek');
+  const endOfWeek = endDay.endOf('isoWeek');
+  const difference = endOfWeek.diff(startOfWeek, 'days') + 1;
+
+  const table = [];
+  let rows = [];
+  for (let i = 0; i < difference; i += 1) {
+    const day = startOfWeek.add(i === 0 ? 0 : 1, 'days').unix();
+    const dayUnix = 24 * 60 * 60;
+    const endOfDay = parseInt(day, 0) + dayUnix;
+
+    let availType = NOT_SPECIFIED;
+    if (day < startTime || day > endTime) {
+      availType = OUT_OF_RANGE;
+    } else {
+      const results = timeSegments.filter(
+        timeSegment => day <= timeSegment.startTime && endOfDay >= timeSegment.endTime,
```

Comment:
woah that's tough to read - confused the shit out of me. Maybe see if the linter will let you do this:
```
timeSegment => (day <= timeSegment.startTime && endOfDay >= timeSegment.endTime),
```

Reply:
The auto-formatter adjusts it to that unfortunately.